mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java

```diff
@@ -88,30 +91,28 @@ public void removeGlobalListener(MailboxListener listener, MailboxSession sessio
 
     @Override
     public void event(Event event) {
-        deliverEventToGlobalListeners(event);
-        if (event instanceof MailboxEvent) {
-            mailboxEvent((MailboxEvent) event);
-        }
-    }
-
-    private void mailboxEvent(MailboxEvent mailboxEvent) {
-        Collection<MailboxListener> listenerSnapshot = registry.getLocalMailboxListeners(mailboxEvent.getMailboxId());
-        if (mailboxEvent instanceof MailboxDeletion && listenerSnapshot.size() > 0) {
-            registry.deleteRegistryFor(mailboxEvent.getMailboxId());
+        ImmutableList<MailboxListener> listeners = ImmutableList.<MailboxListener>builder()
+            .addAll(registry.getGlobalListeners())
+            .addAll(registeredMailboxListeners(event))
+            .build();
+
+        eventDelivery.deliver(listeners, event)
+            .synchronousListenerFuture()
+            .block();
+
+        if (event instanceof MailboxDeletion) {
```

Comment:
why did you remove this condition `listenerSnapshot.size() > 0`? I see it in old version.

Reply:
This seems like a useless optimisation: the cost of removing listeners is low if they do not exists.

I don't want to make the code more complicated for such kind of low-level optimisation. The simpler the conditions are, the better.

Reply:
Ps: Proof that it was useless: I did not broke any tests when removing it :-) #TDD_Refactoring